packages/ra-ui-materialui/src/button/DeleteButton.tsx

```diff
@@ -1,12 +1,40 @@
 import * as React from 'react';
-import { FC, ReactElement, SyntheticEvent } from 'react';
+import { FC, ReactElement, SyntheticEvent, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { Record, RedirectionSideEffect, MutationMode } from 'ra-core';
 
 import { ButtonProps } from './Button';
 import DeleteWithUndoButton from './DeleteWithUndoButton';
 import DeleteWithConfirmButton from './DeleteWithConfirmButton';
 
+const useMutationMode = (mutationMode, undoable) => {
+    useEffect(() => {
```

Comment:
I'm -1 with this useEffect and the related warnings. A simple function, rather than a hook, will be enough.